### PR TITLE
Update some examples using 3.3 features

### PR DESCRIPTION
**Pull request by GuilhemN**
I updated some examples using 3.3 di features, but there is still a lot to do.

### CODE REVIEW

File: best_practices/templates.rst [hunk 1]
```diff
@@ -147,18 +137,21 @@ service in the constructor of the Twig extension:
...
-            tags: [twig.extension]
+        _defaults:
+            # ... be sure autowiring and autoconfiguring are enabled
+            autowire: true
+            autoconfigure: true
+        # ...
+
+        # add Service/ to the list of directories to load services from
+        AppBundle\:
+            resource: '../../src/AppBundle/{Service,Updates,Command,Form,EventSubscriber,Twig,Security}'
```
Comment (xabbuh) [inline best_practices/templates.rst]: the `Updates` namespace looks wrong to me

File: form/create_custom_field_type.rst [hunk 2]
```diff
@@ -303,37 +303,31 @@ the ``genders`` parameter value as the first argument to its to-be-created
...
 
-        # src/AppBundle/Resources/config/services.yml
+        # app/config/services.yml
         services:
-            app.form.type.gender:
-                class: AppBundle\Form\Type\GenderType
+            AppBundle\Form\Type\GenderType:
                 arguments:
                     - '%genders%'
                 tags: [form.type]
```
Comment (xabbuh) [inline form/create_custom_field_type.rst]: What about using `autoconfigure` instead?

Comment (xabbuh) [inline form/create_custom_field_type.rst]: same in other places

Comment (xabbuh) [inline form/create_custom_field_type.rst]: Sounds good

**Comment by GuilhemN** [comment]
I think I'll split this PR in smaller PRs like #7902 because it's much longer to respect https://github.com/symfony/symfony-docs/pull/7877#discussion_r116263681.

**Comment by weaverryan** [comment]
> I updated some examples using 3.3 di features, but there is still a lot to do.

@GuilhemN In fact, I apologize, because I'm completely running over your work :/. There are just so many entries to update, that I've been keeping a list locally and running through them (and making GIANT PRs). That's not idea, but I don't think we'll be able to update everything in time without doing that. However, that *does* mean that those PR's need review (and also, possible some of the most important docs could/should be proofread afterwards ). 

However, you do have a few entires I don't. Could you rebase? So far, I'm skipping the following directories and saving for later:

- best_practices
- bundles
- components
- console
- create_framework
- reference

Thanks!

**Comment by GuilhemN** [comment]
> @GuilhemN In fact, I apologize, because I'm completely running over your work :/. There are just so many entries to update, that I've been keeping a list locally and running through them (and making GIANT PRs).

No problem, this PR didn't take me long to do :)

> Could you rebase?

done ;)

Do you still need help for the remaining parts?

File: form/type_guesser.rst [hunk 3]
```diff
@@ -188,24 +188,23 @@ and tag it with ``form.type_guesser``:
...
         <!-- app/config/services.xml -->
         <?xml version="1.0" encoding="UTF-8" ?>
         <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsd="http://www.w3.org/2001/XMLSchema-instance"
-            xsd:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd"
-        >
-            <services>
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
```
Comment (xabbuh) [inline form/type_guesser.rst]: this change should probably also be made on older branches

File: reference/dic_tags.rst [hunk 4]
```diff
@@ -886,8 +862,7 @@ of your configuration and tag it with ``routing.loader``:
     .. code-block:: yaml
 
         services:
-            app.custom_routing_loader:
-                class: AppBundle\Routing\CustomLoader
+            AppBundle\Routing\CustomLoader:
                 tags: [routing.loader]
```
Comment (xabbuh) [inline reference/dic_tags.rst]: could be merged with the previous line

File: reference/dic_tags.rst [hunk 5]
```diff
@@ -763,8 +745,7 @@ You can add a processor globally:
     .. code-block:: yaml
 
         services:
-            my_service:
-                class: Monolog\Processor\IntrospectionProcessor
+            Monolog\Processor\IntrospectionProcessor:
                 tags: [monolog.processor]
```
Comment (xabbuh) [inline reference/dic_tags.rst]: could be merged with the previous line

File: reference/dic_tags.rst [hunk 6]
```diff
@@ -636,8 +627,7 @@ configuration and tag it with ``kernel.event_subscriber``:
     .. code-block:: yaml
 
         services:
-            app.custom_subscriber:
-                class: AppBundle\EventListener\CustomSubscriber
+            AppBundle\EventListener\CustomSubscriber:
                 tags: [kernel.event_subscriber]
```
Comment (xabbuh) [inline reference/dic_tags.rst]: could be merged with the previous line

**Review by xabbuh — APPROVED** [review]

File: reference/dic_tags.rst (hunk 6 above)
Comment (GuilhemN) [inline reference/dic_tags.rst]: indeed, done

File: form/type_guesser.rst (hunk 3 above)
Comment (GuilhemN) [inline form/type_guesser.rst]: done in https://github.com/symfony/symfony-docs/pull/7915, found many others...

**Comment by xabbuh** [comment]
@GuilhemN Can you resolve the conflicts? :)

**Comment by GuilhemN** [comment]
@xabbuh done

**Comment by weaverryan** [comment]
Thanks @GuilhemN!

**Pull request merged**